### ISSUE TITLE
In Edit on submit data, there is no modal with "Undo" btn except for the first editing

### ISSUE THREAD

**Issue by nurizzzka** [marmelab/react-admin#6901]
**What you were expecting:**
In Edit after saving the data, I wanna see the modal with the "Undo" button for EVERY row, not just for the first editing

**What happened instead:**
After saving data in Edit, there is showing the modal only once

**Steps to reproduce:**
1. In Edit change data on first time there is a modal showing with the 'Undo' button
2. But on the next editing data there is no modal with the 'Undo' button

**Environment**
* React-admin version: 3.19.0
* React version: 16.13.1
* Browser: Chrome (ver 96.0.4664.45 - 64)

**Comment by WiXSL** [comment]
Please provide a sample application showing the issue by forking the following CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple).

**Comment by nurizzzka** [comment]
@WiXSL  With this bug there is no need for the new sample
1. Open https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple
2. Edit any "Post"
3. Save
4. You will see 
![image](https://user-images.githubusercontent.com/39913575/143426515-72e17d31-13cf-4e0f-af84-38ec7678e995.png)
5. Edit one more "Post"
6. Save it
7. It will redirect to the list of "Posts"
**But without the modal with "Undo" option**

**Comment by WiXSL** [comment]
Ok, this seems to be a problem in the simple project, I can't reproduce it in the demo.
Let me investigate further on this.

**Comment by WiXSL** [comment]
This is weird indeed. Happens in tutorial also, but not in the other projects. We look into this.
Marking as a bug

**Comment by WiXSL** [comment]
Note: This doesn't happen in the `next` branch

**Comment by WiXSL** [comment]
Duplicate of #6928

**Comment by fzaninotto** [comment]
Or rather, #6928 is a duplicate of this one, as it's older.

**Comment by WiXSL** [comment]
I traced down this problem to version 3.19.0.
I don't know exactly why this happens, but #6929 seems to fix it

**Comment by nurizzzka** [comment]
@WiXSL I copied the code from #6929, didn't work for me. I'll do further on this

**Comment by WiXSL** [comment]
Did you try react-admin last release (v3.19.3) ?

**Comment by WiXSL** [comment]
I'm closing this issue as fixed by  #6929
If you're still having problems, feel free to re-open it and give details on how to reproduce it.

**Comment by nurizzzka** [comment]
Thanks @WiXSL, after update from 3.19.0 to 3.19.3 the problem was fixed
